File: src/brs/services/TransactionService.java
package brs.services;

import brs.SignumException;
import brs.Transaction;

public interface TransactionService {

  boolean verifyPublicKey(Transaction transaction);

  void validate(Transaction transaction) throws SignumException.ValidationException;
  
  void startNewBlock();

  boolean applyUnconfirmed(Transaction transaction);

  void apply(Transaction transaction);

  void undoUnconfirmed(Transaction transaction);
}

File: src/main/kotlin/brs/services/TransactionService.kt
package brs.services

import brs.entity.Transaction

interface TransactionService {
    /**
     * TODO
     */
    fun verifyPublicKey(transaction: Transaction): Boolean

    /**
     * TODO
     */
    fun validate(transaction: Transaction, preValidate: Boolean = true)

    fun preValidate(transaction: Transaction, height: Int)

    /**
     * TODO
     */
    fun applyUnconfirmed(transaction: Transaction): Boolean

    /**
     * TODO
     */
    fun apply(transaction: Transaction)

    /**
     * TODO
     */
    fun undoUnconfirmed(transaction: Transaction)
}
